draft-ietf-lake-traces.md

```diff
@@ -2817,6 +2838,118 @@ OSCORE Master Salt after KeyUpdate (Raw Value) (8 bytes)
+03 81 02 58 20 74 1a 13 d7 ba 04 8f bb 61 5e 94 38 6a a3 b6 1b ea 5b
+3d 8f 65 f3 26 20 b7 49 be e8 d2 78 ef a9 0e
+~~~~~~~~
+
+Invalid length of the third element (G_X). Selected cipher suite is cipher suite 24 with curve P-384 according to Sections 5.2.2, and 10.2 of {{I-D.ietf-lake-edhoc}}. Correct length of x-coordinate is 48 bytes according to Section 3.7 of {{I-D.ietf-lake-edhoc}} and Section 7.1.1 of {{RFC9053}}.
+
+~~~~~~~~
+Invalid message_1 (40 bytes)
+03 82 02 18 18 58 20 74 1a 13 d7 ba 04 8f bb 61 5e 94 38 6a a3 b6 1b
+ea 5b 3d 8f 65 f3 26 20 b7 49 be e8 d2 78 ef a9 0e
+~~~~~~~~
+
+Invalid type of the third element (G_X). Correct encoding is a byte string according to Section 5.2.1 of {{I-D.ietf-lake-edhoc}}.
+
+~~~~~~~~
+Invalid message_1 (37 bytes)
+03 02 78 20 20 61 69 72 20 73 70 65 65 64 20 6F 66 20 61 20 75 6E 6C
+61 64 65 6E 20 73 77 61 6C 6C 6F 77 20 0e
+~~~~~~~~
+
+Invalid length of third element (G_X). Correct encoding is with leading zeros according to Section 3.7 of {{I-D.ietf-lake-edhoc}} and Section 7.1.1 of {{RFC9053}}.
+
+~~~~~~~~
+Invalid message_1 (36 bytes)
+03 02 58 1f d9 69 77 25 d2 3a 68 8b 12 d1 c7 e0 10 8a 08 c9 f7 1a 85
+a0 9c 20 81 49 76 ab 21 12 22 48 fc 0e
+~~~~~~~~
+
+Invalid x-coordinate in G_X as x {{{≥}}} p. Requirement that x < p according to Section 9.2 of {{I-D.ietf-lake-edhoc}} and Section 5.6.2.3 of {{SP-800-56A}}.
+
+~~~~~~~~
+Invalid message_1 (37 bytes)
+03 02 58 20 ff ff ff ff 00 00 00 01 00 00 00 00 00 00 00 00 00 00 00
+00 ff ff ff ff ff ff ff ff ff ff ff ff 0e
+~~~~~~~~
+
+Invalid x-coordinate in (G_X) not corresponding to a point on the P-256 curve. Requirement that y<sup>2</sup> {{{≡}}} x<sup>3</sup> + a {{{⋅}}} x + b (mod p) according to Section 9.2 of {{I-D.ietf-lake-edhoc}} and Section 5.6.2.3 of {{SP-800-56A}}.
+
+~~~~~~~~
+Invalid message_1 (37 bytes)
+03 02 58 20 a0 4e 73 60 1d f5 44 a7 0b a7 ea 1e 57 03 0f 7d 4b 4e b7
+f6 73 92 4e 58 d5 4c a7 7a 5e 7d 4d 4a 0e
+~~~~~~~~
+
+Invalid number of elements in the CBOR sequence. Correct number of elements is 1 according to Section 5.3.1 of {{I-D.ietf-lake-edhoc}}.
+
+~~~~~~~~
+Invalid message_2 (46 bytes)
+58 20 41 97 01 d7 f0 0a 26 c2 dc 58 7a 36 dd 75 25 49 f3 37 63 c8 93
+42 2c 8e a0 f9 55 a1 3a 4f f5 d5 4B 98 62 a1 1d e4 2a 95 d7 85 38 6a
+~~~~~~~~
+
+Invalid encoding a1 04 42 32 10 of ID_CRED_R in PLAINTEXT_2. Correct encoding is 42 32 10 according to Section 3.5.3.2 of {{I-D.ietf-lake-edhoc}}.
+
+~~~~~~~~
+Invalid PLAINTEXT_2 (15 bytes)
+27 a1 04 42 32 10 48 fa 5e fa 2e bf 92 0b f3
+~~~~~~~~
+
+Invalid encoding 41 32 of ID_CRED_R in PLAINTEXT_2. Correct encoding is 32 according to Section 3.5.3.2 of {{I-D.ietf-lake-edhoc}}.
+
+~~~~~~~~
+Invalid PLAINTEXT_2 (12 bytes)
+27 41 32 48 fa 5e fa 2e bf 92 0b f3
+~~~~~~~~
+
+Invalid encoding of the map ID_CRED_R in PLAINTEXT_2. Correct is the deterministic bytewise lexicographic order encoding according to Section 1.4 of {{I-D.ietf-lake-edhoc}} and Section 4.2.1 of {{RFC8949}}.
+
```

Comment:
Instead of Section 1.4 of -lake-edhoc, I think it is better to refer to both Sections 3.1 (as mentioning deterministic encoding) and Section 3.5.3 (as talking about ID_CRED_X and its encoding).

Note that Section 3.5.3 is not *explicitly* saying that the CBOR map encoding ID_CRED_X has to be deterministically encoded; it is just silently inheriting that requirement from Section 3.1.

Is it worth adding an explicit reminder in Section 3.5.3 in -lake-edhoc? Such a reminder is in fact present in Section 3.5.2 when discussing authentication credentials instead.

Reply:
I will suggst to add an explicit reminder in Section 3.5.3 during AUTH48